exchange/src/wrapper/main.cpp

```diff
@@ -97,6 +107,29 @@ catch_sigint(int)
     while (true) {}
 }
 
+std::optional<std::string>
+get_algorithm()
```

Comment:
This seems like a good opportunity to change comms.hpp because there's a lot of shared functionality. Why don't we have a function that reads std::getline into a templated variant class?

For example, read_mess<market_order, start_time> would return a variant<market_order, start_time>